src/pages/DetailsPage.js

```diff
@@ -108,6 +108,10 @@ class DetailsPage extends React.PureComponent {
             pronouns = this.props.translate(`pronouns.${localeKey}`);
         }
 
+        const phoneNumber = getPhoneNumber(details);
+        const displayName = isSMSLogin ? this.props.formatPhoneNumber(phoneNumber) : details.displayName;
+        const communicationsLinkValue = isSMSLogin ? getPhoneNumber(details) : details.login;
```

Comment:
nitpick: we can make it more readable by naming it based on what it actually stores - `phoneOrEmail`
```suggestion
        const phoneOrEmail = isSMSLogin ? getPhoneNumber(details) : details.login;
```